### ISSUE TITLE
Invalid remainder sending raw transactions (type 2)

### ISSUE THREAD

**Issue by itsdeka** [ConsenSys-archive/ganache#1114]
Hello,
neither building a raw transaction with TypedTransaction nor using [Etherscan API](https://etherscan.io/getRawTx?tx=0x8e3844a10894f466a45e9568f858f5a4a3adbabf8f33adb65648d4c867761e55) I am able to send a raw type 2 transaction on a local fork using ganache-cli.

 ```
w3.eth.send_raw_transaction(HexBytes('0x02f901b4018201f284655ccb42851f4a3cf19a8304ab3294e592427a0aece92de3edee1f18e0157c0586156480b90144c04b8d59000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000a0000000000000000000000000a18479c5fa3306bb5d34c73614118992546405d30000000000000000000000000000000000000000000000000000000061323b6700000000000000000000000000000000000000000000006a082c3898df55024d00000000000000000000000000000000000000000000000000000005e74827bf00000000000000000000000000000000000000000000000000000000000000426b3595068778dd592e39a122f4f5a5cf09c90fe2000bb8c02aaa39b223fe8d0a0e5c4f27ead9083c756cc20001f4a0b86991c6218b36c1d19d4a2e9eb0ce3606eb48000000000000000000000000000000000000000000000000000000000000c080a08180e3965a8dac0fb2674a0b70322727cef7f88dbff690cdf0d926a7caf4db96a06d39fe0a5ffe435c7addbf1b14e90eef4fb0290d5e4f11ec3ee956b29c59dd4b'))

ValueError: {'message': 'invalid remainder', 'code': -32000, 'data': {'stack': 'Error: invalid remainder\n    at Object.t.decode (/usr/local/lib/node_modules/ganache-cli/build/ganache-core.node.cli.js:4:9150)\n    at S.queueRawTransaction (/usr/local/lib/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:210203)\n    at p.eth_sendRawTransaction (/usr/local/lib/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:200119)\n    at p.handleRequest (/usr/local/lib/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:196970)\n    at t (/usr/local/lib/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:65024)\n    at a.handleRequest (/usr/local/lib/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:195252)\n    at t (/usr/local/lib/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:65024)\n    at f.s.handleRequest (/usr/local/lib/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:178304)\n    at f.handleRequest (/usr/local/lib/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:176115)\n    at t (/usr/local/lib/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:65024)\n    at o.handleRequest (/usr/local/lib/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:184369)\n    at t (/usr/local/lib/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:65024)\n    at a.handleRequest (/usr/local/lib/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:183814)\n    at t (/usr/local/lib/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:65024)\n    at u._handleAsync (/usr/local/lib/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:65060)\n    at Timeout._onTimeout (/usr/local/lib/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:64488)', 'name': 'Error'}}

```

Appreciate any help

**Comment by cds-amal** [comment]
Closing as this is fixed in [ganache@7.0.0-alpha.1](https://github.com/trufflesuite/ganache/releases/tag/ganache%407.0.0-alpha.1)

**Comment by andrekreativ** [comment]
What does "invalid remainder" mean? Is there a workaround to send the raw transaction if we are not in ganache@7.0.0-alpha.1? Thank you.

**Comment by davidmurdoch** [comment]
@trackrunner In this case it means ganache-cli@6.*.* doesn't understand what a type 2 transaction is so it can't figure out how to decode the raw transaction data. Use `ganache@7.0.3`.

**Comment by andrekreativ** [comment]
I was using Ganache UI for Windows (latest version 2.5.4 as of today). I switched to "ganache cli" to the version you mentioned (ganache@7.0.3), and send_raw_transaction works fine. @davidmurdoch thanks for the help.

**Comment by benams** [comment]
I'm still having this issue, using ganache@7.7.2 javascript provider. 
Tried 2 different transactions: 
1. https://etherscan.io/tx/0x4ed432c9971962e78700e5e5bea52be58d106e5012b9cfc00d1beb2ce14d03a1
2. https://etherscan.io/tx/0xa50c8224f00d0df312983262f9f62a38caf1439f80125c57ec374366b2ed8db3

```
const provider = ganache.provider({
        fork: { network: "mainnet", blockNumber: 16341084 },
        wallet: {
            unlockedAccounts: [
                "0xa21740833858985e4d801533a808786d3647fb83", // senderAddress,
                "0x5b5a922e065a425ee7e9b70185ba563ef270afcd"
            ]
        },
        // miner: {
        //     defaultGasPrice: 0,
        //     defaultTransactionGasLimit: 0,
        //     callGasLimit: 0
        // }
    });
const rawTransaction =           "0x13d79a0b000000000000000000000000000000000000000000000000000000000000008000000000000000000000000000000000000000000000000000000000000000e0000000000000000000000000000000000000000000000000000000000000014000000000000000000000000000000000000000000000000000000000000003600000000000000000000000000000000000000000000000000000000000000002000000000000000000000000243cacb4d5ff6814ad668c3e225246efa886ad5a000000000000000000000000c02aaa39b223fe8d0a0e5c4f27ead9083c756cc2000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000002540be400000000000000000000000000000000000000000000000000e209f9dcd124e01a00000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000010000000000000000000000003928d9830844f5e58aac234be6c45c68ac1a267200000000000000000000000000000000000000000021db8c156d714072000000000000000000000000000000000000000000000000000000005936a94a5153f30000000000000000000000000000000000000000000000000000000063b6f100e9f29ae547955463ed535162aefee525d8d309571a2b18bc26086c8c35d781eb00000000000000000000000000000000000000000007805d58f6952900000000000000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000021db8c156d71407200000000000000000000000000000000000000000000000000000000000000000001600000000000000000000000000000000000000000000000000000000000000041a401fac3361134a8d2c90ab2df957c777f74891be3625bae0e1849f716c2c0de26982b7c0f5dd301895dd79b02789019223bf70181b0bc3e62fcf43cb153dbfe1b00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000240000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000020000000000000000000000000def171fe48cf0115b1d80b88dc8eab59176fee570000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000e40b86a4c1000000000000000000000000243cacb4d5ff6814ad668c3e225246efa886ad5a00000000000000000000000000000000000000000021db8c156d7140720000000000000000000000000000000000000000000000000000000058d5577661ec90000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000001000000000000000000004de4e351c5517da206948e678018b98f248fa6759e5a000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000";
const rawTransactionResut = await sendRawTransaction(provider, Buffer.from(rawTransaction, 'hex'));
```

yeileds:

```
eth_sendRawTransaction
webpack://Ganache/../../chains/ethereum/ethereum/lib/src/transaction-pool.js:127
                return new ethereum_utils_1.CodedError(ethereum_utils_1.INTRINSIC_GAS_TOO_LOW, utils_2.JsonRpcErrorCode.INVALID_INPUT);
                       ^
CodedError: intrinsic gas too low
    at TransactionPool.call (webpack://Ganache/../../chains/ethereum/ethereum/lib/src/transaction-pool.js:127:24)
    at TransactionPool.prepareTransaction (webpack://Ganache/../../chains/ethereum/ethereum/lib/src/transaction-pool.js:146:87)
    at TransactionManager.add (webpack://Ganache/../../chains/ethereum/ethereum/lib/src/data-managers/transaction-manager.js:108:48)
    at Blockchain.queueTransaction (webpack://Ganache/../../chains/ethereum/ethereum/lib/src/blockchain.js:939:55)
    at EthereumApi.apply (webpack://Ganache/../../chains/ethereum/ethereum/lib/src/api.js:1717:27)
    at EthereumApi.apply (webpack://Ganache/../../chains/ethereum/ethereum/lib/src/helpers/assert-arg-length.js:14:28)
    at Object.execute (webpack://Ganache/../utils/lib/src/utils/request-coordinator.js:80:47)
    at RequestCoordinator.call (webpack://Ganache/../utils/lib/src/utils/request-coordinator.js:59:22)
    at webpack://Ganache/../utils/lib/src/utils/request-coordinator.js:89:80
    at new Promise (<anonymous>)
    at RequestCoordinator.queue (webpack://Ganache/../utils/lib/src/utils/request-coordinator.js:76:20)
    at Executor.execute (webpack://Ganache/../utils/lib/src/utils/executor.js:68:92)
    at EthereumProvider._requestRaw (webpack://Ganache/../../chains/ethereum/ethereum/lib/src/provider.js:259:92)
    at EthereumProvider.request (webpack://Ganache/../../chains/ethereum/ethereum/lib/src/provider.js:247:38)
    at sendRawTransaction (/Users/benamsalem/proj/fireblocks/transaction-simulator/src/ganache.ts:268:35)
    at Subscription.<anonymous> (/Users/benamsalem/proj/fireblocks/transaction-simulator/src/ganache.ts:57:43) {
  code: -32000
}
```

Any ideas what I can change?

**Comment by davidmurdoch** [comment]
@benams This looks like a completely different error to me. Can you open your own issue for this?

**Comment by davidmurdoch** [comment]
@benams I tried to reproduce, but can't with your example, as I you didn't provide the implementation for `sendRawTransaction`. But I did look up the original raw transaction (https://etherscan.io/getRawTx?tx=0x4ed432c9971962e78700e5e5bea52be58d106e5012b9cfc00d1beb2ce14d03a1) and ran that successfully. Ganache command I used: `ganache --fork mainnet --fork.blockNumber 16341084 --wallet.unlockedAccounts 0xa21740833858985e4d801533a808786d3647fb83 --chain.chainId=1 --chain.time="Jan-05-2023 02:08:11 PM UTC" --miner.timestampIncrement=13`.
